pkg/tsdb/cloud-monitoring/annotation_query.go

```diff
@@ -38,6 +38,7 @@ func (s *Service) executeAnnotationQuery(ctx context.Context, req *backend.Query
 		} `json:"timeSeriesList"`
 	}{}
 
+	// It's okay to use the first query for annotations as there should only be one
```

Comment:
more seriously... if you're sure you only get a single Query here, you could add something like this to be safe:

```go
	if len(req.Queries) != 1 {
		return nil, errors.New("multiple queries received in annotation-request")
	}
```

Reply:
Good call out, thank you!